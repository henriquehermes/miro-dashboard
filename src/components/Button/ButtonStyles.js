import styled, { css } from 'styled-components';

const defaultFont = css`
  color: ${props => props.theme.secondary};
  font-weight: bold;
`;

const marginHorizontal = css`
  margin: 0px 5px;
`;

const defaultBorder = css`
  border-radius: 6px;
  border-color: ${props => props.theme.primary};
  border-style: solid;
`;

const defaultBox = css`
  border-radius: 6px;
  border-color: #fff;
  border-style: solid;
  box-shadow: 0px 4px 18px 0px rgba(0, 0, 0, 0.1);
`;

const defaultActive = css`
  background: ${props => props.theme.primary};
  color: #fff;
`;

const Button = styled.button`
  cursor: pointer;
  height: 40px;
  width: 90px;
  ${defaultFont}
  background: #fff;
  ${props => (props.withBorder ? defaultBorder : defaultBox)};
  ${props => (props.marginHorizontal ? marginHorizontal : '')};
  ${props => (props.disabled ? 'opacity: 0.5' : 1)};
  ${props => (props.active ? defaultActive : '')};

  @media (max-width: 767px) {
    ${props => (props.marginVertical ? `margin-top: 10px; width: 100%;` : '')}
  }
`;

export default Button;
