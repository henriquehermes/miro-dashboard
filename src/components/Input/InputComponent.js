/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';

import Input from './InputStyles';

export default function InputComponent(props) {
  return <Input {...props} />;
}
