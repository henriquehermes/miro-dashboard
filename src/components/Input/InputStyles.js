import styled, { css } from 'styled-components';

const defaultFont = css`
  color: ${props => props.theme.secondary};
  font-weight: bold;
`;

const defaultBox = css`
  border-radius: 6px;
  border-color: #fff;
  border-style: solid;
  box-shadow: 0px 4px 18px 0px rgba(0, 0, 0, 0.1);
`;

const Input = styled.input`
  height: 40px;
  width: 100%;
  margin: 10px 0;
  padding: 0 10px;
  border: none;
  background: #fff;
  ${props => (props.hasBox ? `${defaultBox}` : '')};
  ${defaultFont}
`;

export default Input;
