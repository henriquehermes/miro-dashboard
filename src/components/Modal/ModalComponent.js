/* eslint-disable react/forbid-prop-types */
import React, { useState, useEffect } from 'react';
import Modal from 'react-responsive-modal';
import PropTypes from 'prop-types';

import { putData, postData } from "../../services/DashboardService"

import Form  from './ModalStyles';

export default function ModalComponent({
  data,
  isVisible,
  handleClose,
  updateBooks,
}) {
  const [book, setBook] = useState({
    title: '',
    author: '',
    pages: '',
    year: '',
  });

  useEffect(() => {
    if (data) {
      setBook({
        title: data.title,
        author: data.author,
        pages: data.pages,
        year: data.year,
      });
    }
  }, [data]);

  async function handleSubmit(e) {
    e.preventDefault();
    const { title, author, pages, year } = book;

    if (title !== '' && author !== '' && pages !== '' && year !== '') {
      book.rented = false;
      const response = await postData(book)
      updateBooks(response);
      handleClose();
    }
  }

  async function handleEdit(e) {
    e.preventDefault();
    const { title, author, pages, year } = book;

    if (title !== '' && author !== '' && pages !== '' && year !== '') {
      const response = await putData(book, data.id)
      updateBooks(response, true);
      handleClose();
    }
  }

  return (
    <Modal open={isVisible} onClose={handleClose} center closeIconSize={25}>
      <h2>
        {data ? 'Editar' : 'Adicionar'}
        {' '}
        livro
      </h2>
      <Form>
        <input
          placeholder='Titulo'
          id='title'
          type='text'
          value={book.title}
          onChange={event => setBook({ ...book, title: event.target.value })}
        />
        <input
          placeholder='Autor'
          id='author'
          type='text'
          value={book.author}
          onChange={event => setBook({ ...book, author: event.target.value })}
        />
        <input
          placeholder='Paginas'
          id='pages'
          type='text'
          value={book.pages}
          onChange={event => setBook({ ...book, pages: event.target.value })}
        />
        <input
          placeholder='Ano'
          id='year'
          type='text'
          value={book.year}
          onChange={event => setBook({ ...book, year: event.target.value })}
        />
        {data ? (
          <button type="button" onClick={e => handleEdit(e)}>Salvar</button>
        ) : (
          <button type="button" onClick={e => handleSubmit(e)}>Adicionar</button>
        )}
      </Form>
    </Modal>
  );
}

ModalComponent.propTypes = {
  data: PropTypes.shape.isRequired,
  isVisible: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  updateBooks: PropTypes.func.isRequired,
};
