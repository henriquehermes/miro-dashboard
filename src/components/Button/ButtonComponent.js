import React from 'react';

import { Button } from './ButtonStyles';

export default function ButtonComponent(props) {
  return <Button {...props} />;
}
