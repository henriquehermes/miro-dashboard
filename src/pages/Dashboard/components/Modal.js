import React, { useState, useEffect } from 'react';
import Modal from 'react-responsive-modal';
import api from '../../../config/api';

import { Form } from './styles';

export default function ModalComponent({
  data,
  isVisible,
  handleClose,
  updateBooks,
}) {
  const [book, setBook] = useState({
    title: '',
    author: '',
    pages: '',
    year: '',
  });

  useEffect(() => {
    if (data) {
      setBook({
        title: data.title,
        author: data.author,
        pages: data.pages,
        year: data.year,
      });
    }
  }, [data]);

  async function handleSubmit(e) {
    e.preventDefault();
    const { title, author, pages, year } = book;

    if (title !== '' && author !== '' && pages !== '' && year !== '') {
      book.rented = false;
      const { data } = await api.post(`/books`, book);
      updateBooks(data);
      handleClose();
    }
  }

  async function handleEdit(e) {
    e.preventDefault();
    const { title, author, pages, year } = book;

    if (title !== '' && author !== '' && pages !== '' && year !== '') {
      const response = await api.put(`/books/${data.id}`, book);

      updateBooks(response.data, true);
      handleClose();
    }
  }

  return (
    <Modal open={isVisible} onClose={handleClose} center closeIconSize={25}>
      <h2>{data ? 'Editar' : 'Adicionar'} livro</h2>
      <Form>
        <input
          placeholder='Titulo'
          id='title'
          type='text'
          value={book.title}
          onChange={event => setBook({ ...book, title: event.target.value })}
        />
        <input
          placeholder='Autor'
          id='author'
          type='text'
          value={book.author}
          onChange={event => setBook({ ...book, author: event.target.value })}
        />
        <input
          placeholder='Paginas'
          id='pages'
          type='text'
          value={book.pages}
          onChange={event => setBook({ ...book, pages: event.target.value })}
        />
        <input
          placeholder='Ano'
          id='year'
          type='text'
          value={book.year}
          onChange={event => setBook({ ...book, year: event.target.value })}
        />
        {data ? (
          <button onClick={e => handleEdit(e)}>Salvar</button>
        ) : (
          <button onClick={e => handleSubmit(e)}>Adicionar</button>
        )}
      </Form>
    </Modal>
  );
}
