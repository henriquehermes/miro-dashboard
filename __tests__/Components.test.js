import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Input from '../src/components/Input/InputComponent';
import Button from '../src/components/Button/ButtonComponent';
import Modal from '../src/components/Modal/ModalComponent';

describe('Input', () => {
  test('Remder Input', () => {
    const component = renderer.create(<Input />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Button', () => {
  test('Remder Button', () => {
    const component = renderer.create(<Button />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Modal', () => {
  test('Remder Modal', () => {
    const component = renderer.create(<Modal />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
