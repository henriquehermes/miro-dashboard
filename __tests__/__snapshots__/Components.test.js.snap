// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Remder Button 1`] = `
.c0 {
  cursor: pointer;
  height: 40px;
  width: 90px;
  font-weight: bold;
  background: #fff;
  border-radius: 6px;
  border-color: #fff;
  border-style: solid;
  box-shadow: 0px 4px 18px 0px rgba(0,0,0,0.1);
}

<button
  className="c0"
/>
`;

exports[`Input Remder Input 1`] = `
.c0 {
  height: 40px;
  width: 100%;
  margin: 10px 0;
  padding: 0 10px;
  border: none;
  background: #fff;
  font-weight: bold;
}

<input
  className="c0"
/>
`;

exports[`Modal Remder Modal 1`] = `null`;
